src/org/labkey/test/util/SampleSetHelper.java

```diff
@@ -312,6 +313,16 @@ public void bulkImport(List<Map<String, String>> data, String importOption, @Nul
         }
     }
 
+    public SampleSetHelper deleteSamples(String expectedTitle)
+    {
+        getSamplesDataRegionTable().clickHeaderButton("Delete");
+        Window.Window(getDriver()).withTitle(expectedTitle).waitFor()
+                .clickButton("Yes, Delete", true);
```

Comment:
I was able to reproduce locally the failure for SampleSetTest.testDeleteSamplesSomeWithDerivedSamples that is happening on TeamCity. I was able to get it to pass if I changed the last parameter of the clickButton call to false. 
What is happening is that this clickButton is waiting until the window closes before it returns. That's causing the next line (withTitleContaining("Delete sample").waitFor() to fail the expected condition. 

Maybe you don't need the these two lines:
Window.Window(getDriver()).withTitleContaining("Delete sample").waitFor();
 _ext4Helper.waitForMaskToDisappear();

It looks like this is what is also causing the SampleSetTest.testDeleteSamplesSomeWithAssayData test to fail as well.

Reply:
Thanks, Dan.  changing the "true" to "false" did the trick.